### PR TITLE
DYN-6313 Add ML Node AutoComplete TOU

### PR DESCRIPTION
**Pull request by jesusalvino**
### Purpose

Implementing the improvement https://jira.autodesk.com/browse/DYN-6313

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Reviewers
@QilongTang

### FYIs
@Jingyi-Wen 
@RobertGlobant20 
@Enzo707

### CODE REVIEW

File: src/DynamoCoreWpf/Properties/Resources.en-US.resx [hunk 1]
```diff
@@ -2217,7 +2217,10 @@ Installing the latest update requires Dynamo and any host applications to close.
 Do you want to install the latest Dynamo update?</value>
   </data>
   <data name="UsageReportPromptDialogTitle" xml:space="preserve">
-    <value>Agreement to Collect Usability Data</value>
+    <value>Agreement and Term of Use</value>
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Properties/Resources.en-US.resx:2220]: Term(s)? spelling

File: src/DynamoCoreWpf/Properties/Resources.en-US.resx [hunk 2]
```diff
@@ -2217,7 +2217,10 @@ Installing the latest update requires Dynamo and any host applications to close.
...
+  </data>
+  <data name="IAgreeToMLAutocompleteTOU" xml:space="preserve">
+    <value>I agree to ML Node Autocomplete Term of Use</value>
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Properties/Resources.en-US.resx:2223]: Terms

**Comment by mjkkirschner** [comment]
worth checking with someone else, but I believe all the instances of `term` should be `terms`.

**Comment by jesusalvino** [comment]
> worth checking with someone else, but I believe all the instances of `term` should be `terms`.

sure, Just following the Mockup enclosed in the task, that's why I'm tagging @Jingyi-Wen

**Comment by QilongTang** [comment]
hi @jesusalvino Isn't there a toggle similar to the analytics one in your gif? We need to record the value if user agreed on it or not in preferences.

**Comment by jesusalvino** [comment]
> hi @jesusalvino Isn't there a toggle similar to the analytics one in your gif? We need to record the value if user agreed on it or not in preferences.

Done.

File: src/DynamoCoreWpf/Properties/Resources.en-US.resx (hunk 2 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/Properties/Resources.en-US.resx]: Thanks @mjkkirschner I updated across Dynamo code base

**Comment by QilongTang** [comment]
@avidit Any idea why the UI Smoke test would fail?

**Comment by avidit** [comment]
> @avidit Any idea why the UI Smoke test would fail?

It is because of the change in `UsageReportPromptDialogTitle`, I have a task to update the test.

**Comment by QilongTang** [comment]
@avidit After the fix on master branch, is it fine for me to merge this one for now?

**Comment by avidit** [comment]
> @avidit After the fix on master branch, is it fine for me to merge this one for now?

Yes

**Pull request merged**